area/area_test.go

```diff
@@ -39,6 +39,45 @@ func (test *TestAreaRepository) TearDownTest() {
 	test.clean()
 }
 
+func (test *TestAreaRepository) TestCreateAreaWithSameNameFail() {
+	t := test.T()
+
+	resource.Require(t, resource.Database)
+
+	repo := area.NewAreaRepository(test.DB)
+
+	name := "Area 21"
+	newSpace := space.Space{
+		Name: "Space 1 " + uuid.NewV4().String(),
+	}
+	repoSpace := space.NewRepository(test.DB)
+	space, err := repoSpace.Create(context.Background(), &newSpace)
+	assert.Nil(t, err)
+
+	i := area.Area{
+		Name:    name,
+		SpaceID: space.ID,
+	}
+
+	repo.Create(context.Background(), &i)
+	if i.ID == uuid.Nil {
+		t.Errorf("Area was not created, ID nil")
+	}
+
+	if i.CreatedAt.After(time.Now()) {
+		t.Errorf("Area was not created, CreatedAt after Now()?")
+	}
+
+	assert.Equal(t, name, i.Name)
+
+	anotherAreaWithSameName := area.Area{
+		Name:    i.Name,
+		SpaceID: space.ID,
+	}
+	err = repo.Create(context.Background(), &anotherAreaWithSameName)
+	assert.NotNil(t, err) // Is there a way to check error code?
```

Comment:
Good point,
I've used an assertion on the error type using your suggestion. Thanks. 

https://github.com/almighty/almighty-core/pull/913/commits/d9db51cd06fb3464eaec7ff832e3ce36e955bbd5#diff-69df85e87042165a4b16f9b12717e69cR80